### ISSUE TITLE
R-drop makes my model broken.

### ISSUE THREAD

**Issue by MayDomine** [dropreg/R-Drop#12]
In my NMT task,I try to let the encoder and decoder to forward twice ,but the kl_loss is too large.
Then I tried to compute the mean,but it is too small to have effect.
![image](https://user-images.githubusercontent.com/57244158/129010041-ee2743c9-966f-497d-bcca-294086beeb67.png)
![image](https://user-images.githubusercontent.com/57244158/129010135-a058cb97-af7e-4121-a02b-ad99f851e7a2.png)

Can someone help me？

**Comment by dropreg** [comment]
@MayDomine may be some mistake in line "loss = ce_loss + kl_loss   loss = lambda_coeff  * loss", and you can try to following code: 
> loss = ce_loss + lambda_coeff * kl_loss 

At the same time,  if  kl_loss is still abnormal, you can just sum up the last dimension:
> p_loss = p_loss.sum(-1).mean()

**Comment by MayDomine** [comment]
@dropreg I tried as you recommend.And now my loss reduce normally,but the bleu score is decreasing.
any suggestion？
maybe I should try to add R-drop when pretraing?
now I just add it when finetuning.

**Comment by MayDomine** [comment]
And I noticed someone said their kl-loss decreased very fast.And my kl-loss seems to decrease quite slow.

**Comment by dropreg** [comment]
@MayDomine  It's worth noting that both Dropout Rate and lambda_coeff are very important hyperparameters, and R-Drop based method may take longer to train.  Make sure the training epoch is larger than baseline !

**Comment by apeterswu** [comment]
Close it since there are no further comments. Reopen it if needed.

**Comment by XiaoqingNLP** [comment]
@MayDomine Hi, did you have improvement on your task? do you remember the ratio about the values of ce and kl? did you try to add R-drop in pretraining?

**Comment by MayDomine** [comment]
&nbsp;&nbsp;you need to train with more epoches,try set ratio 0.25&nbsp;



---Original---
From: ***@***.***&gt;
Date: Wed, Feb 16, 2022 15:35 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [dropreg/R-Drop] R-drop makes my model broken. (#12)




 
@MayDomine Hi, did you have improvement on your task? Do you remember the ratio about the values of ce and kl?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by XiaoqingNLP** [comment]
@MayDomine Hi, What does the ratio in your reply mean? This kl loss is very small compared to ce. In addition, this loss drop is so erratic that I wonder if there is a problem.

**Comment by MayDomine** [comment]
I met this question too, and my advice is that you should set the ratio to make kl-loss is much smaller than loss of model.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "dropreg/R-Drop"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月17日(星期四) 中午1:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [dropreg/R-Drop] R-drop makes my model broken. (#12)





 
@MayDomine Hi, What does the ratio in your reply mean? This kl loss is very small compared to ce. In addition, this loss drop is so erratic that I wonder if there is a problem.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
